SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,823 @@
+
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+
+    if (geometry->nodes->GetDomain(iPoint)) {
+      /*--- Set enthalpy on the wall ---*/
+
+      prog_wall = solver_container[SPECIES_SOL]->GetNodes()->GetSolution(iPoint)[I_PROGVAR];
+      if (config->GetSpecies_StrongBC()) {
+        /*--- Initial guess for enthalpy value ---*/
+
+        enth_init = nodes->GetSolution(iPoint, I_ENTH);
+        enth_wall = enth_init;
+
+        n_not_iterated += fluid_model_local->GetEnthFromTemp(&enth_wall, prog_wall, temp_wall, enth_init);
+
+        /*--- Impose the value of the enthalpy as a strong boundary
+        condition (Dirichlet) and remove any
+        contribution to the residual at this node. ---*/
+
+        nodes->SetSolution(iPoint, I_ENTH, enth_wall);
+        nodes->SetSolution_Old(iPoint, I_ENTH, enth_wall);
+
+        LinSysRes(iPoint, I_ENTH) = 0.0;
+
+        nodes->SetVal_ResTruncError_Zero(iPoint, I_ENTH);
+
+        if (implicit) {
+          unsigned long total_index = iPoint * nVar + I_ENTH;
+          Jacobian.DeleteValsRowi(total_index);
+        }
+      } else {
+        /*--- Weak BC formulation ---*/
+        const auto Normal = geometry->vertex[val_marker][iVertex]->GetNormal();
+
+        const su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+        const auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+        /*--- Get coordinates of i & nearest normal and compute distance ---*/
+
+        const auto Coord_i = geometry->nodes->GetCoord(iPoint);
+        const auto Coord_j = geometry->nodes->GetCoord(Point_Normal);
+        su2double Edge_Vector[MAXNDIM];
+        GeometryToolbox::Distance(nDim, Coord_j, Coord_i, Edge_Vector);
+        su2double dist_ij_2 = GeometryToolbox::SquaredNorm(nDim, Edge_Vector);
+        su2double dist_ij = sqrt(dist_ij_2);
+
+        /*--- Compute the normal gradient in temperature using Twall ---*/
+
+        su2double dTdn = -(flowNodes->GetTemperature(Point_Normal) - temp_wall) / dist_ij;
+
+        /*--- Get thermal conductivity ---*/
+
+        su2double thermal_conductivity = flowNodes->GetThermalConductivity(iPoint);
+
+        /*--- Apply a weak boundary condition for the energy equation.
+        Compute the residual due to the prescribed heat flux. ---*/
+
+        LinSysRes(iPoint, I_ENTH) -= thermal_conductivity * dTdn * Area;
+      }
+    }
+  }
+  if (rank == MASTER_NODE && n_not_iterated > 0) {
+    cout << " !!! Isothermal wall bc (" << Marker_Tag
+         << "): Number of points in which enthalpy could not be iterated: " << n_not_iterated << " !!!" << endl;
+  }
+}
+
+void CSpeciesFlameletSolver::BC_ConjugateHeat_Interface(CGeometry* geometry, CSolver** solver_container,
+                                                        CNumerics* conv_numerics, CConfig* config,
+                                                        unsigned short val_marker) {
+
```

Comment:
Duplicated isothermal boundary...

Reply:
merged in the same way as for NSSolver...